docs/content/src/learn/guides/services/converting-json-to-xml-and-upload-to-ftp/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Prerequisites
+Link to download ballerina integrator.
+
+## Implementation
+The ballerina project should create for the integration use case explained above. Please follow the steps given below. You can learn about the ballerina project and module in this link.
+
+1. Create a project
+```bash
+$ ballerina new converting-json-to-xml-and-upload-to-ftp
+```
+Change directory to the converting-json-to-xml-and-upload-to-ftp.
+
+2. Add a module
+```bash
+$ ballerina add uploadtoftp
+```
+
+The project structure should look like below.
+```shell
+converting-json-to-xml-and-upload-to-ftp
+├── Ballerina.toml
+└── src
+    └── uploadtoftp
+        ├── main.bal
+        ├── Module.md
+        ├── resources
+        └── tests
+            └── resources
+```
+
+3. Write the integration
+You can open the project with VSCode. The integration implemention going to write in the `main.bal` file. 
+
+<!-- INCLUDE_CODE: src/uploadtoftp/main.bal -->
+
+The `ftpConfig` created providing file server credentials. Here it reads from the config file. You have to create a file called `ballerina.conf` under the root path of the project structure. The file should have the following configurations.
+
+```
+[ftp]
+  host="<file_server_url>"
+  port=<file_server_port>
+  username="<file_server_username>"
+  password="<file_server_password>"
+```
+
+The username and password are sensitive data. Usually, we don't expose those data to the outside. The ballerina supports encrypting sensitive data and uses them in the program.
+
+```shell
+$ ballerina encrypt
+```
+
+The execution of the encrypt action will ask for value and secret key. Once you provide input, it'll output the encrypted value that can directly use in the config file.
+
+```shell
+Enter value: 
+
+Enter secret: 
+
+Re-enter secret to verify: 
+
+Add the following to the configuration file:
+<key>="@encrypted:{aoIlSvOPeBEZ0COma+Wz2uWznlNn1IWz4StiWQCO6g4=}"
+
+Or provide it as a command line argument:
+--<key>=@encrypted:{aoIlSvOPeBEZ0COma+Wz2uWznlNn1IWz4StiWQCO6g4=}
+```
+
+Use encrypted value in the `ballerina.conf` file.
+
+```
+[ftp]
+  host="<file_server_url>"
+  port=<file_server_port>
+  username="<file_server_username>"
+  password="@encrypted:{3SCS/ET4uaJKVZwh+/If3rvU9ImbZSCmrruroMF5Sig=}"
+```
+
+After that `ftp:Client` created with `ftpConfig`. The HTTP listener service will run on 8080 port when you start running the program. When request receives to the HTTP service, get the JSON payload and transformed to XML. Since ballerina language is type-safe language, we have to check transformed result returns XML or error. If XML returned, then we are writing the XML to the remote FTP location specified in the top with the FTP client. The return result of the `ftp:put` operation was successful, then respond to the call building JSON payload using `http:Response.` If there is any error return from the above actions, then relevant error message builds with the error reason and respond to the caller. 
```

Comment:
```suggestion
After that `ftp:Client` is created with `ftpConfig`. The HTTP listener service will run on 8080 port when you start running the program. When the request is received by the HTTP service, it gets the JSON payload and transforms it to XML. Since Ballerina language is a type-safe language, we have to check if the transformed result returns XML or error. If XML is returned, then we write the XML to the remote FTP location specified in the top with the FTP client. If the return result of the `ftp:put` operation was successful, respond to the call by building JSON payload using `http:Response`. If any error is returned from the above actions, the relevant error message builds with the error reason and responds to the caller. 
```